Packages/DesignSystem/Sources/DesignSystem/Views/AvatarView.swift

```diff
@@ -2,104 +2,315 @@ import Nuke
 import NukeUI
 import Shimmer
 import SwiftUI
+import Models
 
 @MainActor
 public struct AvatarView: View {
-  @Environment(\.redactionReasons) private var reasons
   @Environment(Theme.self) private var theme
 
-  public enum Size {
-    case account, status, embed, badge, list, boost
+  @State private var showPopup = false
+  @State private var autoDismiss = true
+  @State private var toggleTask: Task<Void, Never> = Task {}
 
-    public var size: CGSize {
-      switch self {
-      case .account:
-        return .init(width: 80, height: 80)
-      case .status:
-        if ProcessInfo.processInfo.isMacCatalystApp {
-          return .init(width: 48, height: 48)
-        }
-        return .init(width: 40, height: 40)
-      case .embed:
-        return .init(width: 34, height: 34)
-      case .badge:
-        return .init(width: 28, height: 28)
-      case .list:
-        return .init(width: 20, height: 20)
-      case .boost:
-        return .init(width: 12, height: 12)
+  public let account: Account?
+  public let config: FrameConfig
+  public let hasPopup: Bool
+
+  public var body: some View {
+    if let account = account {
+      if hasPopup {
+        AvatarImage(account: account, config: adaptiveConfig)
+          .onHover { hovering in
+            if hovering {
+              toggleTask.cancel()
+              toggleTask = Task {
+                try? await Task.sleep(nanoseconds: NSEC_PER_SEC / 2)
```

Comment:
Too bad we don't have a proper SwiftUI way to delay the hover.

Reply:
Luckily, We have Swift concurrency. 😁